cmd/zed/zed.d/data-notify.sh

```diff
@@ -25,7 +25,7 @@ zed_rate_limit "${rate_limit_tag}" || exit 3
 
 umask 077
 note_subject="ZFS ${ZEVENT_SUBCLASS} error for ${ZEVENT_POOL} on $(hostname)"
-note_pathname="${TMPDIR:="/tmp"}/$(basename -- "$0").${ZEVENT_EID}.$$"
+note_pathname="$(mktemp)"
```

Comment:
`mktemp zed.XXXXXXXXXX`?

Reply:
I don't really see the point in deviating from the default for a file that's `rm -f`ed like 15 lines below in all cases (also, it'd be `mktemp -t zed.XXXXXXXXXX`)

Reply:
The only advantage I can see is it makes it easy to identify these zed file if they somehow fail to get cleaned up.  The existing manual scheme seems fine, but I don't have any objection to just using `mktemp`.